### PR TITLE
Issue #13345: Enable examples tests for CyclomaticComplexityCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for CyclomaticComplexityCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **CyclomaticComplexityCheck** in the project. The implementation focuses on ensuring that CyclomaticComplexity checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
@romani  Is different examples fine for this check?

**Comment by romani** [comment]
Please make Examples to be same by structure

**Comment by AmitKumarDeoghoria** [comment]
> Please make Examples to be same by structure

since the examples are long in length making them same by structure will cause the file length to exceed 60 lines and cause violation. Should I suppress the file length violation for this check?

**Comment by romani** [comment]
https://www.diffchecker.com/0KdwtPkY/

1 and 3 are almost same.
Please remove empty lines between empty methods. If files still above limits, please suppress.

**Comment by AmitKumarDeoghoria** [comment]
Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — CHANGES_REQUESTED** [review]
Items to shrink files in size with hope to avoid suppression

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          break;
+        default:
+          break;
+      }
+    }
+    a = a > 0 ? b : c;
+  }
+
+  private void fun5() {}
+
```
[the remark this answers is not in the document]
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <property name="tokens" value="LITERAL_WHILE, LITERAL_DO"/>
+    </module>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.metrics.cyclomaticcomplexity;
+
+// xdoc section -- start
+class Example2 {
+  // Cyclomatic Complexity = 5
```
[the remark this answers is not in the document]
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  int a, b, c, d, e, n;
+  // violation below, 'Cyclomatic Complexity is 5 (max allowed is 4)'
+  public void testMethod1() { // 1, function declaration
+    while (a < b // 2, while
+      && a > c) {
```
[the remark this answers is not in the document]
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/cyclomaticcomplexity/Example2.java]: Done

**Comment by romani** [comment]
GitHub, generate website

File: config/checkstyle-examples-suppressions.xml [hunk 4]
```diff
@@ -62,6 +62,10 @@
   <suppress checks="CommentsIndentation"
     files="commentsindentation[\\/]Example[1-9].java"/>
 
+  <!-- violation required by to show behavior of Check -->
+  <suppress checks="FileLength"
+    files="cyclomaticcomplexity[\\/]Example\d+.*"/>
```
Comment (romani) [inline config/checkstyle-examples-suppressions.xml:67]: Please remove this, your examples below 60 line limit now

**Review by romani — CHANGES_REQUESTED** [review]
Last

File: config/checkstyle-examples-suppressions.xml (hunk 4 above)
Comment (AmitKumarDeoghoria) [reply config/checkstyle-examples-suppressions.xml:67]: I think It checks the length of the entire file not just the example so we need the suppression.
Screenshot 2024-11-20 at 5 16 30 AM

Comment (romani) [reply config/checkstyle-examples-suppressions.xml:67]: Ok, in this case let's keep suppression

**Pull request merged**